sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcSourceSuite.scala

```diff
@@ -332,6 +334,23 @@ abstract class OrcSuite extends OrcTest with BeforeAndAfterAll {
       assert(version === SPARK_VERSION_SHORT)
     }
   }
+
+  test("SPARK-26859 Reading ORC files with explicit schema can result in wrong data") {
```

Comment:
does this test help the test coverage? It seems is same as the `AddColumnIntoTheMiddleTest`